File: app/src/main/java/com/amicly/alamofoursquare/common/scheduler/TrampolineSchedulerProvider.java
package com.amicly.alamofoursquare.common.scheduler;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by darrankelinske on 2/1/18.
 */

public class TrampolineSchedulerProvider implements SchedulerProvider {
    @Override
    public Scheduler io() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler computation() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler mainThread() {
        return Schedulers.trampoline();
    }
}

File: app/src/main/java/com/amicly/alamofoursquare/common/scheduler/TrampolineSchedulerProvider.kt
package com.amicly.alamofoursquare.common.scheduler

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

/**
 * Created by darrankelinske on 2/1/18.
 */

class TrampolineSchedulerProvider : SchedulerProvider {
    override fun io(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun computation(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun mainThread(): Scheduler {
        return Schedulers.trampoline()
    }
}
